Cubical/HITs/CumulativeHierarchy/Base.agda

```diff
@@ -0,0 +1,672 @@
+    isPropPathP : (i : I) → isProp (PathP (λ j → B i j) (r i) (s i))
+    isPropPathP i = isProp→isSet-PathP (λ j → isPropB i j) (r i) (s i)
+
+  transport⁻ua : ∀ {A B : Type ℓ} (e : A ≃ B) → transport⁻ (ua e) ≡ transport (ua (invEquiv e))
+  transport⁻ua e = EquivJ (λ _ e → transport⁻ (ua e) ≡ transport (ua (invEquiv e))) refl e
+
+module EmbeddingsExt a {b} (B : Type b) where
+  open import Cubical.Functions.Fibration as Fibr
+  open import Cubical.Foundations.Equiv.Fiberwise as Fiber
+  -- an embedding from some type into B
+  AnEmbedding : Type (ℓ-max (ℓ-suc a) b)
+  AnEmbedding = Σ[ A ∈ Type a ] Σ[ f ∈ (A → B) ] isEmbedding f
+  _↪ : (e : AnEmbedding) → e .fst → B
+  _↪ = fst ∘ snd
+
+  Fiberwise→Path : (f g : AnEmbedding)
+                 → (∀ b → fiber (f ↪) b → fiber (g ↪) b)
+                 → (∀ b → fiber (g ↪) b → fiber (f ↪) b)
+                 → f ≡ g
+  Fiberwise→Path f g f→g g→f = λ i → ua F≡G i , f≡g i , ef≡eg i where
+    propFibersF : {b : B} → isProp (fiber (f ↪) b)
+    propFibersF = isEmbedding→hasPropFibers (f .snd .snd) _
+    propFibersG : {b : B} → isProp (fiber (g ↪) b)
+    propFibersG = isEmbedding→hasPropFibers (g .snd .snd) _
+    fiberIso : (b : B) → fiber (f ↪) b ≃ fiber (g ↪) b
+    fiberIso _ = isoToEquiv (iso (f→g _) (g→f _) (λ _ → propFibersG _ _) (λ _ → propFibersF _ _))
+    FiberEquiv : Σ B (fiber (f ↪)) ≃ Σ B (fiber (g ↪))
+    FiberEquiv = _ , Fiber.totalEquiv (fiber (f ↪)) (fiber (g ↪)) (fst ∘ fiberIso) (snd ∘ fiberIso)
+    F≡G : f .fst ≃ g .fst
+    F≡G = f .fst            ≃⟨ Fibr.totalEquiv _ ⟩
+          Σ B (fiber (f ↪)) ≃⟨ FiberEquiv ⟩
+          Σ B (fiber (g ↪)) ≃⟨ invEquiv (Fibr.totalEquiv _) ⟩
+          g .fst ■
+    f≡g : PathP (λ i → ua F≡G i → B) (f ↪) (g ↪)
+    f≡g = toPathP (funExt λ x → λ i → transportRefl (invEquiv FiberEquiv .fst (Fibr.totalEquiv (g ↪) .fst (transportRefl x i)) .snd .snd i) i)
+    ef≡eg : PathP (λ i → isEmbedding (f≡g i)) (f .snd .snd) (g .snd .snd)
+    ef≡eg = isProp→PathP (λ i → isEmbeddingIsProp {f = f≡g i}) _ _
+
+isEmbeddingLiftsIsProp : ∀ {a b} {A : Type a} {B : Type b} {f : A → B} → isEmbedding f
+                       → isProp B → isProp A
+isEmbeddingLiftsIsProp {f = f} embeds isPropB x y = invEquiv (cong f , embeds x y) .fst (isPropB (f x) (f y))
+
+expandΣinΠ : ∀ {a b c}{A : Type a}{B : A → Type b}{C : ∀ a → B a → Type c}
+           → (∀ a → (b : B a) → C a b) ≃ (((a , b) : Σ A B) → C a b)
+expandΣinΠ = isoToEquiv (iso (λ f (a , b) → f a b) (λ f a b → f (a , b)) (λ f → refl) (λ f → refl))
+
+------------
+-- Start implementation of V
+------------
+
+-- set up the basic hierarchy definition and _∈_ as recursive, higher inductive types
+data V (ℓ : Level) : Type (ℓ-suc ℓ)
+_∈_ : (S T : V ℓ) → hProp (ℓ-suc ℓ)
+
+eqImage : {X Y : Type ℓ} (ix : X → V ℓ) (iy : Y → V ℓ) → Type (ℓ-suc ℓ)
+eqImage {X = X} {Y = Y} ix iy = (∀ (a : X) → ∥ Σ[ b ∈ Y ] (iy b) ≡ (ix a) ∥) ⊓′ (∀ (b : Y) → ∥ Σ[ a ∈ X ] (ix a) ≡ (iy b) ∥)
```

Comment:
You can use \exists for truncated Sigma: https://github.com/agda/cubical/blob/master/Cubical/Data/Sigma/Base.agda#L31